### PR TITLE
[Layout Foundation][Box] Add support for passing in role attribute

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Supports #7797 which needs access to role attribute 

### WHAT is this pull request doing?

Adds role attribute support to `Box`

### CODE REVIEW

**Comment by chazdean** [comment]
Decided to add role manually as a Type rather then extending `React.HTMLAttributes<HTMLElement>`
- This way we don't open up `Box` to all HTML attributes
- This also allows more control as status, presentation and combobox are they only other roles currently used in web

**Review by aveline — DISMISSED** [review]
I think you are missing `role` in `createElement` so it actually gets applied to the element?

**Comment by aveline** [comment]
For context, just wanted to add that we are creating the list of allowable roles manually to intentionally discourage the use of role when there are suitable semantic html elements that should be used instead.

Ex.
❌ `<div role="nav">`
✅ `<nav>`

File: polaris-react/src/components/Box/Box.tsx [hunk 1]
```diff
@@ -150,6 +151,8 @@ export interface BoxProps extends React.AriaAttributes {
    * paddingInlineEnd={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
    */
   paddingInlineEnd?: Spacing;
+  /** role of box */
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.tsx]:
```suggestion
  /** Aria role */
```

File: polaris-react/src/components/Box/Box.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ import styles from './Box.scss';
 type Element = 'div' | 'span' | 'section';
 type Overflow = 'hidden' | 'scroll';
 type Position = 'relative' | 'absolute' | 'fixed' | 'sticky';
+type Role = 'status';
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.tsx]:
```suggestion
```

File: polaris-react/src/components/Box/Box.tsx [hunk 3]
```diff
@@ -150,6 +151,8 @@ export interface BoxProps extends React.AriaAttributes {
    * paddingInlineEnd={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
    */
   paddingInlineEnd?: Spacing;
+  /** role of box */
+  role?: Role;
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.tsx]: Let's keep these based the types we know are a11y related

```suggestion
  role?: Extract<React.AriaRole, 'status'>;
```

We'll probably allow all or most of what's avaliable in `AriaRole` but for now we can be strict

**Review by kyledurand — COMMENTED** [review]
Looking good but I think we can be a bit more intentional about what this prop should be used for 👍

File: polaris-react/src/components/Box/Box.tsx (hunk 3 above)
Comment (chazdean) [reply polaris-react/src/components/Box/Box.tsx]: I like this 👌🏾

**Review by kyledurand — APPROVED** [review]
✨

**Pull request merged**